### PR TITLE
Add Gov messages for validator deny list.

### LINKED ISSUE #1327: Add Validator Deny List add/remove proposals
## Summary

The validator deny list logic was added in #1268, but we need a mechanism for governance to add and remove validators from this list.

This is in the form of a message type (see MsgGovCloseChannel in ics, or the MsgIncentiveSpend in airdrop) which requires a title, a description, an authority (in place of the fromAddress) and whatever fields this particular instance requires.

I would suggest two messages, one for add and one for remove. The following fields ought to be sufficient.

Fields:
- chain_id
- operator_address

### PR DESCRIPTION
**Pull request by tropicaldog**
## 1. Summary
Fixes #1327

## 2.Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## 5. Checklist

- [ ] Does the Readme need to be updated?

## Summary by CodeRabbit

- **New Features**
	- Introduced governance actions for managing validator deny lists in the interchain staking module, allowing for the addition and removal of validators from deny lists.
- **Tests**
	- Added tests for governance-related operations on validator deny lists.
- **Documentation**
	- Updated documentation related to the new governance actions and message types.

### CODE REVIEW

**Comment by joe-bowman** [comment]
This pull request has been deployed to Vercel.

<table>
<tr>
<td><strong>Latest commit:</strong></td>
<td><code>5dd127a</code></td>
</tr>
<tr>
<td><strong>✅ Preview:</strong></td>
<td><a href='https://pr-1329.app-dev.quicksilver.zone'>https://pr-1329.app-dev.quicksilver.zone</a></td>
</tr>
<tr>
<td><strong>🔍 Inspect:</strong></td>
<td><a href='https://vercel.com/quicksilver/quicksilver/FcSKczr5xYDVjbDC252mXJHMQSLH'>https://vercel.com/quicksilver/quicksilver/FcSKczr5xYDVjbDC252mXJHMQSLH</a></td>
</tr>
</table>

[View Workflow Logs](https://github.com/quicksilver-zone/quicksilver/actions/runs/8434441034)

File: proto/quicksilver/interchainstaking/v1/messages.proto [hunk 1]
```diff
@@ -109,3 +123,29 @@ message MsgSignalIntent {
 
 // MsgSignalIntentResponse defines the MsgSignalIntent response type.
 message MsgSignalIntentResponse {}
+
+// MsgAddValidatorDenyList represents a message type for adding validator to deny list
+message MsgAddValidatorDenyList {
+  option (gogoproto.equal) = false;
+  option (gogoproto.goproto_getters) = false;
+
+  string chain_id = 1 [(gogoproto.moretags) = "yaml:\"chain_id\""];
```
Comment (joe-bowman) [inline proto/quicksilver/interchainstaking/v1/messages.proto:132]: These require a title and description. They actually serve no functional purpose, but all block explorer expect these fields to be present in order to display to the user.

Comment (joe-bowman) [reply proto/quicksilver/interchainstaking/v1/messages.proto:132]: Also, from_address should be `authority`

File: proto/quicksilver/interchainstaking/v1/messages.proto [hunk 2]
```diff
@@ -109,3 +123,29 @@ message MsgSignalIntent {
...
+  string operator_address = 2 [(gogoproto.moretags) = "yaml:\"chain_id\""];
+  string from_address = 3 [(cosmos_proto.scalar) = "cosmos.AddressString"];
+}
+
+// MsgAddValidatorDenyListResponse defines the MsgSignalIntent response type.
+message MsgAddValidatorDenyListResponse {}
+
+// MsgRemoveValidatorDenyList represents a message type for removing validator from deny list
+message MsgRemoveValidatorDenyList {
+  option (gogoproto.equal) = false;
```
Comment (joe-bowman) [inline proto/quicksilver/interchainstaking/v1/messages.proto:142]: As above.

File: proto/quicksilver/interchainstaking/v1/messages.proto [hunk 3]
```diff
@@ -109,3 +123,29 @@ message MsgSignalIntent {
...
+  option (gogoproto.goproto_getters) = false;
+
+  string chain_id = 1 [(gogoproto.moretags) = "yaml:\"chain_id\""];
+  string operator_address = 2 [(gogoproto.moretags) = "yaml:\"chain_id\""];
+  string from_address = 3 [(cosmos_proto.scalar) = "cosmos.AddressString"];
+}
+
+// MsgRemoveValidatorDenyListResponse defines the MsgSignalIntent response type.
+message MsgRemoveValidatorDenyListResponse {}
```
[the remark this answers is not in the document]
Comment (joe-bowman) [reply proto/quicksilver/interchainstaking/v1/messages.proto:151]: ^^ this.

File: proto/quicksilver/interchainstaking/v1/messages.proto (hunk 1 above)
Comment (joe-bowman) [reply proto/quicksilver/interchainstaking/v1/messages.proto:132]: Oh I see this exists in proposals. MsgAddValidatorDenyList and MsgRemoveValidatorDenyList shouldn't exist then.

File: proto/quicksilver/interchainstaking/v1/proposals.proto [hunk 4]
```diff
@@ -127,3 +127,33 @@ message MsgGovSetLsmCaps {
...
+
+message MsgGovRemoveValidatorDenyList {
+  option (gogoproto.equal) = false;
+  option (gogoproto.goproto_getters) = false;
+
+  string title = 1;
+  string description = 2;
+
+  string chain_id = 3 [(gogoproto.moretags) = "yaml:\"chain_id\""];
+  string operator_address = 4 [(gogoproto.moretags) = "yaml:\"chain_id\""];
```
Comment (joe-bowman) [inline proto/quicksilver/interchainstaking/v1/proposals.proto:154]:
```suggestion
  string operator_address = 4 [(gogoproto.moretags) = "yaml:\"operator_address\""];
```

File: proto/quicksilver/interchainstaking/v1/proposals.proto [hunk 5]
```diff
@@ -127,3 +127,33 @@ message MsgGovSetLsmCaps {
...
+
+message MsgGovAddValidatorDenyList {
+  option (gogoproto.equal) = false;
+  option (gogoproto.goproto_getters) = false;
+
+  string title = 1;
+  string description = 2;
+
+  string chain_id = 3 [(gogoproto.moretags) = "yaml:\"chain_id\""];
+  string operator_address = 4 [(gogoproto.moretags) = "yaml:\"chain_id\""];
```
Comment (joe-bowman) [inline proto/quicksilver/interchainstaking/v1/proposals.proto:139]:
```suggestion
  string operator_address = 4 [(gogoproto.moretags) = "yaml:\"operator_address\""];
```

File: proto/quicksilver/interchainstaking/v1/messages.proto (hunk 3 above)
Comment (joe-bowman) [inline proto/quicksilver/interchainstaking/v1/messages.proto:151]:
```suggestion
```

These types need not exist,

**Review by joe-bowman — CHANGES_REQUESTED** [review]
Couple of fixes, but functionally looks good

File: proto/quicksilver/interchainstaking/v1/messages.proto (hunk 3 above)
[the remark this answers is not in the document]
Comment (tropicaldog) [reply proto/quicksilver/interchainstaking/v1/messages.proto]: good catch @coderabbitai, tysm!

**Comment by joe-bowman** [comment]
This pull request has been deployed to Vercel.

<table>
<tr>
<td><strong>Latest commit:</strong></td>
<td><code>ec21376</code></td>
</tr>
<tr>
<td><strong>✅ Preview:</strong></td>
<td><a href='https://pr-1329.app-dev.quicksilver.zone'>https://pr-1329.app-dev.quicksilver.zone</a></td>
</tr>
<tr>
<td><strong>🔍 Inspect:</strong></td>
<td><a href='https://vercel.com/quicksilver/quicksilver/94V9iM9AvCcWXG1MBuPVe5uuaviq'>https://vercel.com/quicksilver/quicksilver/94V9iM9AvCcWXG1MBuPVe5uuaviq</a></td>
</tr>
</table>

[View Workflow Logs](https://github.com/quicksilver-zone/quicksilver/actions/runs/8445634407)

**Comment by joe-bowman** [comment]
This pull request has been deployed to Vercel.

<table>
<tr>
<td><strong>Latest commit:</strong></td>
<td><code>c355890</code></td>
</tr>
<tr>
<td><strong>✅ Preview:</strong></td>
<td><a href='https://pr-1329.app-dev.quicksilver.zone'>https://pr-1329.app-dev.quicksilver.zone</a></td>
</tr>
<tr>
<td><strong>🔍 Inspect:</strong></td>
<td><a href='https://vercel.com/quicksilver/quicksilver/7axGKQKPfV1RqVGnYCkcKn7Li4Ro'>https://vercel.com/quicksilver/quicksilver/7axGKQKPfV1RqVGnYCkcKn7Li4Ro</a></td>
</tr>
</table>

[View Workflow Logs](https://github.com/quicksilver-zone/quicksilver/actions/runs/8445672089)

**Review by joe-bowman — APPROVED** [review]
looks great, thnak you

**Pull request merged**